### PR TITLE
Add Javadoc comment which specifies that Realm.deleteRealmFile(...) throws IllegalStateException if target Reealm file is not closed.

### LINKED ISSUE #1107: Document that Realm.deleteRealmFile(...) can throw IllegalStateException
Javadoc does not specify that `Realm.deleteRealmFile(...) ` throws IllegalStateException if target realm is not closed.

### PR DESCRIPTION
**Pull request by zaki50**
Fixes #1107

### CODE REVIEW

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -1643,6 +1643,8 @@ public static synchronized void migrateRealmAtPath(String realmPath, byte[] key,
      * @param context an Android {@link android.content.Context}.
      * @return false if a file could not be deleted. The failing file will be logged.
      * @see io.realm.Realm#clear(Class)
+     *
+     * @throws IllegalStateException if trying to delete a Realm that is already open.
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/Realm.java]: Please prefix with java.lang to avoid ambiguity.

File: realm/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -1655,6 +1657,8 @@ public static boolean deleteRealmFile(Context context) {
      * @param context  an Android {@link android.content.Context}.
      * @param fileName the name of the custom Realm (i.e. "myCustomRealm.realm").
      * @return false if a file could not be deleted. The failing file will be logged.
+     *
+     * @throws IllegalStateException if trying to delete a Realm that is already open.
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/Realm.java]: Please prefix with java.lang to avoid ambiguity.

File: realm/src/main/java/io/realm/Realm.java [hunk 3]
```diff
@@ -1666,6 +1670,8 @@ public static boolean deleteRealmFile(Context context, String fileName) {
      *
      * @param realmFile The reference to the Realm file.
      * @return false if a file could not be deleted. The failing file will be logged.
+     *
+     * @throws IllegalStateException if trying to delete a Realm that is already open.
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/Realm.java]: Please prefix with java.lang to avoid ambiguity.

**Comment by cmelchior** [comment]
Apart from comments. Thanks for fixing this :+1: 

@emanuelez @kneth ?

**Comment by kneth** [comment]
@zaki50 Thanks

**Comment by cmelchior** [comment]
Superseded by #1113

**Comment by zaki50** [comment]
@cmelchior #1108 should be merged because #1113 is for `compactRealmFile()` not for `deleteRealmFile()`

**Comment by kishikawakatsumi** [comment]
@cmelchior @emanuelez @finnschiermer This fix is the same content of https://github.com/realm/realm-java/pull/1113, but for the different methods. It seems to be ready to merge.

**Comment by kneth** [comment]
:+1:

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**